### ISSUE TITLE
Container restart loses connectivity to backends

### ISSUE THREAD

**Issue by phoolish** [IBM/varnish-operator#67]
When Kubernetes restarts the container due to a liveliness probe failure the container comes back with 0 backends.

```
varnishadm backend.list
Backend name   Admin      Probe    Health     Last change
boot.default   healthy    0/0      healthy    Sun, 08 May 2022 02:30:49 GMT
```

I confirmed that  `/etc/varnish/backends.vcl` is still populated correctly and other pods can still connect to the backends without a problem. Deleting the pod "fixes" it.

Here is our VarnishCluster manifest for context.
```
apiVersion: caching.ibm.com/v1alpha1
kind: VarnishCluster
metadata:
  labels:
    operator: varnish
  name: pcms-api
spec:
  backend:
    port: 80
    selector:
      app: pcms
      component: web
      purpose: api
  replicas: 3
  service:
    annotations:
      prometheus.io/path: /metrics
      prometheus.io/port: "9131"
      prometheus.io/scrape-only: "true"
    port: 80
  varnish:
    args:
    - -p
    - http_max_hdr=256
    - -p
    - http_resp_hdr_len=256k
    - -p
    - http_resp_size=1024k
    - -p
    - workspace_backend=256k
    - -s
    - malloc,756M
    resources:
      limits:
        cpu: 500m
        memory: 1028Mi
      requests:
        cpu: 500m
        memory: 1028Mi
  vcl:
    configMapName: pcms-varnishcluster
    entrypointFileName: default.vcl
```

**Comment by cin** [comment]
Thanks for reporting the issue. We'll take a look. FYI: @tomashibm

**Comment by cin** [comment]
Was able to reproduce by `/sbin/killall5` which killed all the processes and containers but still caused the condition mentioned above to occur. Deleting pod fixed it but going to try forcing an update.

**Comment by phoolish** [comment]
@cin Thank you for the the followup. If we need manual intervention it is easier for my team to just delete the pod. 

I wonder if it is restarting and just using the system default vcl `/etc/default/varnish`.

**Comment by tomashibm** [comment]
fix is released in the latest [0.31.0](https://github.com/IBM/varnish-operator/releases/tag/0.31.0) version
